File: WebApplication/src/modules/application/application-one/models/application-one-credit-select.dto.ts
import { AutoMap } from "@automapper/classes";
import { NomenclatureDto } from "src/infrastructure/models/nomenclature.dto";
import { CourseType } from "../enums/course-type.enum";
import { YearType } from "../enums/year-type.enum";
import { CreditSelectDto } from "../../common/models/credit-select.dto";
import { ExistingEnum } from "../../common/enums/existing.enum";

export class ApplicationOneCreditSelectDto extends CreditSelectDto {
  @AutoMap()
  nationality: NomenclatureDto;
  @AutoMap()
  secondNationality: NomenclatureDto;
  @AutoMap()
  idNumber: string;
  @AutoMap()
  researchArea: NomenclatureDto;
  @AutoMap()
  educationFormType: NomenclatureDto;
  @AutoMap()
  educationalQualification: NomenclatureDto;
  @AutoMap()
  studentCourse: CourseType | null;
  @AutoMap()
  doctoralYear: YearType | null;

  @AutoMap()
  contactPerson: string;

  @AutoMap()
  expirationDateOfGracePeriod: Date | null;
  @AutoMap()
  schoolRemaining: number | null;
  @AutoMap()
  creditSize: number | null;
  @AutoMap()
  semesterTax: number | null;

  @AutoMap()
  monthlyPayment: number | null;
  @AutoMap()
  paymentDescription: string;
}

File: StudentCredit.Application/Applications/ApplicationsOne/Dtos/ApplicationOneCreditSelectDto.cs
using StudentCredit.Application.Applications.Common.Dtos;
using StudentCredit.Data.Applications.ApplicationOne.Enums;
using StudentCredit.Data.Rdpzsd.Enums;
using StudentCredit.Infrastructure.Helpers.Dtos;

namespace StudentCredit.Application.Applications.ApplicationsOne.Dtos
{
	public class ApplicationOneCreditSelectDto : CreditSelectDto
	{
		public NomenclatureDto Nationality { get; set; }
		public NomenclatureDto SecondNationality { get; set; }
		public string IdNumber { get; set; }
		public NomenclatureDto ResearchArea { get; set; }
		public NomenclatureDto EducationFormType { get; set; }
		public NomenclatureDto EducationalQualification { get; set; }
		public CourseType? StudentCourse { get; set; }
		public YearType? DoctoralYear { get; set; }

		public string ContactPerson { get; set; }

		public DateTime? ExpirationDateOfGracePeriod { get; set; }
		public int? SchoolRemaining { get; set; }
		public decimal? CreditSize { get; set; }
		public decimal? SemesterTax { get; set; }

		public decimal? MonthlyPayment { get; set; }
		public string PaymentDescription { get; set; }
    }
}
